src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -205,4 +209,73 @@ public boolean isSelected() {
             return selected;
         }
     }
+
+    //=================================================================
+
+    public List<File> getList() throws IOException {
+        BundleBrowser bundleBrowser = new BundleBrowser();
+        return bundleBrowser.getZipFileList();
+    }
+
+    public ListBoxModel doFillGoalTypeItems() throws IOException {
+        ListBoxModel items = new ListBoxModel();
+        List<File> fileList = new ArrayList<File>();
+        BundleBrowser bundleBrowser = new BundleBrowser();
+        File[] list = bundleBrowser.getFileList();
+
+        for(File file:list){
+            fileList.add(file);
+            items.add(file.getName(),file.getName());
+        }
+        return items;
+    }
+
+    public void doDownloadBundle(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        ArrayList<Integer> selectedFileIndices = new ArrayList<Integer>();
+        if (!"POST".equals(req.getMethod())) {
```

Comment:
You can use the `@RequirePOST` annotation instead of the check for if it is not a `POST`.